### PR TITLE
Add version exclude pattern

### LINKED ISSUE #47: Add version exclude pattern
In our use case, we only want to exclude those packages of us from consolidation, that are alpha versions.

For example, we have the [Helpful Libraries](https://www.nuget.org/packages/Lombiq.HelpfulLibraries/) package what is a utility package widely used in our other projects. We might add some new features to it just for OtherProject. Then, we release Helpful Libraries as an alpha (like v8.0.3-alpha.0.lmbq-249) and reference that in OtherProject, of which we release an alpha too. However, since Helpful Libraries is used in dozens of other projects, we are forced to update the reference there too.

This is useful if we release a new non-alpha version of Helpful Libraries, since then we indeed want to bring up all of our projects to the same level. However, this is unnecessary for alpha versions.

Do you think this would be a useful feature, @olsh? If yes, I'd give implementing it a shot, following the pattern of https://github.com/olsh/dotnet-consolidate/pull/25.

### PR DESCRIPTION
**Pull request by Piedone**
Fixes #47

### CODE REVIEW

**Comment by Piedone** [comment]
How can I test this, @olsh? Is it possible for you to publish an alpha to NuGet?

**Comment by Piedone** [comment]
Uhm, should Sonar really fail at this point? Coverage is not actually 0, I added tests for the new feature. And for `Options` I see no use in adding a unit test for a property assignment.

**Comment by olsh** [comment]
> How can I test this, @olsh? Is it possible for you to publish an alpha to NuGet?

You can build the package, add a local folder as a Nuget source, and install the tool.

> build nuget-pack

This command should create a Nuget package.

**Comment by olsh** [comment]
On second thought. Probably we should use existing parameters?
We already have `-p` and `-e`. We could use an array of regexes instead of strings.

Of course, this is a breaking change and we should bump the major version.

Let me know what you think.

**Comment by olsh** [comment]
> Uhm, should Sonar really fail at this point? Coverage is not actually 0, I added tests for the new feature. And for `Options` I see no use in adding a unit test for a property assignment.

Just ignore it. I've just disabled the coverage for the quality gate. The next build should pass.

**Comment by Piedone** [comment]
-p and -e are about package IDs while this is about versions (there's a separate issue about package ID wildcards too). So, I think a new option is adequate.

I'll try local testing like that, thanks.

**Comment by olsh** [comment]
> -p and -e are about package IDs while this is about versions (there's a separate issue about package ID wildcards too). So, I think a new option is adequate.

Oh, yeah 😅. Then yes, let's create a new parameter `--exclude-version-pattern` or something like that.

**Comment by Piedone** [comment]
I named `excludedVersionsRegex` following the patterns of `directoryBuildProps` and `reportOverridenDirectoryBuildProps`. Are you sure you want it to be renamed to dashed format?

File: src/DotNet.Consolidate/Models/Options.cs [hunk 1]
```diff
@@ -6,23 +6,27 @@ namespace DotNet.Consolidate.Models
...
             PackageIds = packageIds;
             ExcludedPackageIds = excludedPackageIds;
-            Solutions = solutions;
+            ExcludedPackageVersionsRegex = excludedPackageVersionsRegex;
             ReadDirectoryBuildProps = readDirectoryBuildProps;
             ReportOverridenDirectoryBuildProps = reportOverridenDirectoryBuildProps;
         }
 
+        [Option('s', "solutions", Required = false, HelpText = "Target solutions for checking. If not specified, all solutions in the working directory will be analyzed.")]
+        public ICollection<string>? Solutions { get; }
```
Comment (Piedone) [inline src/DotNet.Consolidate/Models/Options.cs:20]: Without the order of the properties and ctor parameters being the same one gets the below exception:

```
Unhandled exception. System.InvalidOperationException: Type DotNet.Consolidate.Models.Options appears to be immutable, but no constructor found to accept values.
   at CommandLine.Core.InstanceBuilder.BuildImmutable[T](Type typeInfo, Maybe`1 factory, IEnumerable`1 specProps, IEnumerable`1 specPropsWithValue, List`1 setPropertyErrors)
   at CommandLine.Core.InstanceBuilder.<>c__DisplayClass1_0`1.<Build>b__5()
   at CommandLine.Core.InstanceBuilder.Build[T](Maybe`1 factory, Func`3 tokenizer, IEnumerable`1 arguments, StringComparer nameComparer, Boolean ignoreValueCase, CultureInfo parsingCulture, Boolean autoHelp, Boolean autoVersion, Boolean allowMultiInstance, IEnumerable`1 nonFatalErrors)
   at CommandLine.Parser.ParseArguments[T](IEnumerable`1 args)
   at DotNet.Consolidate.Program.Main(String[] args) in E:\Projects\Lombiq\dotnet-consolidate\src\DotNet.Consolidate\Program.cs:line 26
```

See: https://github.com/commandlineparser/commandline/wiki/Immutable%20Options%20Type#:~:text=Invalid%20constructors

**Comment by olsh** [comment]
> I named `excludedVersionsRegex` following the patterns of `directoryBuildProps` and `reportOverridenDirectoryBuildProps`. Are you sure you want it to be renamed to dashed format?

Ok, let's keep `excludedVersionsRegex` but I still think that the `-v` shorthand is not intuitive and controversial.
Because `-p` (packages) includes packages, and -v (version) excludes versions.

Let's remove the short name and merge the PR. 

Thank you for the contribution 👍🏻

**Comment by Piedone** [comment]
Sure, done that. I just thought that, following the pattern of the other options, having a shorthand switch is desired.

Thank you for being so responsive!

**Pull request merged**